File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> num_list) {
    int answer = 0;

    for (int i = 0; i < num_list.size(); i++) {
        while (num_list[i] != 1) {
            if (num_list[i] % 2 == 0) {
                num_list[i] /= 2;
            } else {
                num_list[i] = (num_list[i] - 1) / 2;
            }
            answer++;
        }
    }

    return answer;
}

File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.swift
import Foundation

func solution(_ num_list:[Int]) -> Int {
    
    var result = 0
    
    num_list.forEach {
        var num = $0
        
        while num != 1 {
            if num.isMultiple(of: 2) {
                num = num / 2
            } else {
                num = (num - 1) / 2
            }
            result += 1
        }
    }
    
    return result
}